pwndbg/commands/heap.py

```diff
@@ -182,6 +182,53 @@ def heap(addr=None, verbose=False, simple=False) -> None:
         for chunk in h:
             malloc_chunk(chunk.address, verbose=verbose, simple=simple)
 
+parser = argparse.ArgumentParser(
+    formatter_class=argparse.RawTextHelpFormatter,
+    description="""Try to find if an address belonds to a heap chunk. If yes -- print chunk. Search all heaps.""",
+)
+parser.add_argument(
+    "addr",
+    type=int,
+    help="Address of the interest.",
+)
+parser.add_argument(
+    "-v", "--verbose", action="store_true", help="Print all chunk fields, even unused ones."
+)
+parser.add_argument(
+    "-s", "--simple", action="store_true", help="Simply print malloc_chunk struct's contents."
+)
+parser.add_argument(
+    "-f", "--fast", action="store_true", help="Be fast and not accurate, or be slow and accurate"
+)
+
+@pwndbg.commands.ArgparsedCommand(parser, category=CommandCategory.HEAP)
+@pwndbg.commands.OnlyWhenRunning
+@pwndbg.commands.OnlyWithResolvedHeapSyms
+@pwndbg.commands.OnlyWhenHeapIsInitialized
+def hi(addr, verbose=False, simple=False, fast=False) -> None:
+    """Iteratively search all heaps for contain current address in their chunks bodys.
+    """
+    allocator = pwndbg.heap.current
+    sbrk_region = allocator.get_sbrk_heap_region()
+    if fast or addr in sbrk_region:
+        heap = Heap(addr)
+        for chunk in heap:
+            if chunk.real_size and chunk.address <= addr and (chunk.address + chunk.real_size) > addr:
+                malloc_chunk(chunk.address, verbose=verbose, simple=simple)
+                break
+        return
+
+    # this is the only solution I was able to found, which *DOES NOT* return false-positive results
```

Comment:
But it can produce false *negative* results when the main arena has become non-contiguous.
The `Heap` class can take care of this logic without instantiating a bunch of other unnecessary `Heap` objects. You can check that `heap.arena is not None` if you don't want the region your user has provided an address in to be treated as a heap when it isn't.

Reply:
Hmmm, I'll check it soon

Reply:
Ah don't sweat it, there's an edge case either way and I think the way it is now will work for most people's use cases.